### DISCUSSION TITLE
Cartographer Role

### DISCUSSION THREAD

**Discussion by Canela352** [dbarzin/mercator#947] [General]
Hello 

I hope you are doing well.
Someone could please provide me details about the role of the "Cartographer".
The guide describe what is user, Auditeur and Admin but not cartographer. 

Thank you

**Comment by lynch4fr** [comment]
Bonjour,
Petit bug technique sur ce sujet "Cartographer"
J'ai ajouté ce role Cartographer à mon utilisateur et il devrait avoir tous les droits sur les applications en theorie.
Et bien curieusement, il n'a pas le bouton "Modifier" dans la "Vue des applications" -> "Applications" alors qu'il est disponible pour tous les autres items de la vue ?
Merci ( _et  :clap: bravo pour votre article dans le Linux Pratique 146_ )

**Reply by dbarzin** [reply]
il faut également modifier une variable dans le fichier de configuration `.env`

    # Uncomment this line if you want to use the Cartographers
    #
    # Cartographers are attached to applications, and, when this enabled, only them (and admins) can modify their own apps
    # and what is attach to (under the application view). Documentation here :
    # CARTOGRAPHERS=true

Alors, lorsqu'un utilisateur a le rôle "cartographer" et qu'il est cartographe de l'application, il peut la modifier.

**Reply by dbarzin** [reply]
![image](https://github.com/user-attachments/assets/963799ef-79ad-470f-a6e1-f19b5c4d7af1)

**Reply by lynch4fr** [reply]
Merci, ça repond tout à fait à notre besoin, dommage que ce role doit etre supprimé :(

**Reply by dbarzin** [reply]
Ce n'est pas une décision ferme. Peux-tu décrire ton besoin, le contexte d'utilisation ?

**Reply by lynch4fr** [reply]
L'idée est vraiment bien et nous voulions justement déléguer la chaîne d'une application, ses services, ses modules, BD et flux à un.e tiers responsable.

Si ce droit devait être supprimé, je me demande comment nous pourrions le remplacer en terme de "rôle" ?

_Au passage, je remarque qu'un cartographe d'un application à aussi les droits R/W sur tous les items (services,modules,BD, flux) même s'ils sont en dehors de son périmètre. Est ce explicable ? Ne devrait il pas y avoir aussi une limitation naturelle sur les items associés entre cartographe et applications+items ?_

Merci

**Reply by dbarzin** [reply]
L'idée était bien celle-là, avoir une gestion des droits que ne soit pas que par type d'objet, mais également pour une application et ses objets liés. Il s'avère que c'est compliqué à gérer en termes de droits d'accès et que la création d'objets liés à une application peut causer des incohérences.

**Comment by dbarzin** [comment]
The "Cartographer" role was introduced experimentally by some contributors. The intention behind this role was to allow a user to update information for a single application and its related objects. However, over time, it became evident that this approach was not practical, and the role is no longer in use. We are planning to remove this feature in a future update.
